<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Zhan Yinxiu" />


<title>Genotype phenotype sAML1</title>

<script src="site_libs/header-attrs-2.24/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<script src="site_libs/plotly-binding-4.10.2/plotly.js"></script>
<script src="site_libs/typedarray-0.1/typedarray.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="site_libs/plotly-main-2.11.1/plotly-latest.min.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.29/datatables.js"></script>
<link href="site_libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<script src="site_libs/jszip-1.13.4/jszip.min.js"></script>
<script src="site_libs/pdfmake-1.13.4/pdfmake.js"></script>
<script src="site_libs/pdfmake-1.13.4/vfs_fonts.js"></script>
<link href="site_libs/dt-ext-buttons-1.13.4/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.13.4/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.4/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.4/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.4/js/buttons.print.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">scmseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://www.research.ieo.it/research-and-technology/technological-units/data-science-unit/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Genotype phenotype sAML1</h1>
<h4 class="author">Zhan Yinxiu</h4>
<h4 class="date">23 October, 2024</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-10-23
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>phenotype_genotype/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240315code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240315)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240315code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240315)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
session-info-chunk-inserted-by-workflowr
</li>
<li>
unnamed-chunk-1
</li>
<li>
unnamed-chunk-2
</li>
<li>
unnamed-chunk-3
</li>
<li>
unnamed-chunk-4
</li>
<li>
unnamed-chunk-5
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory
<code>02_integration_phenotype_genotype_sAML1_cache</code> and re-run
the analysis. To have workflowr automatically delete the cache directory
prior to building the file, set <code>delete_cache = TRUE</code> when
running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/hpcnfs/scratch/P_DIMA_SCMSEQ/analysis/scrnaseq/phenotype_genotype/src/functions.R
</td>
<td style="text-align:left;">
src/functions.R
</td>
</tr>
<tr>
<td style="text-align:left;">
/hpcnfs/scratch/P_DIMA_SCMSEQ/analysis/scrnaseq/phenotype_genotype/data/enrichment/sAML1.csv
</td>
<td style="text-align:left;">
data/enrichment/sAML1.csv
</td>
</tr>
<tr>
<td style="text-align:left;">
/hpcnfs/scratch/P_DIMA_SCMSEQ/analysis/scrnaseq/phenotype_genotype/data/01_tme_annotation.rds
</td>
<td style="text-align:left;">
data/01_tme_annotation.rds
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomieoresearchphenotypegenotypetree8cb6dcdfb22add097a49b0b66c7b4308a2ea96c6targetblank8cb6dcda">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/ieoresearch/phenotype_genotype/tree/8cb6dcdfb22add097a49b0b66c7b4308a2ea96c6" target="_blank">8cb6dcd</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomieoresearchphenotypegenotypetree8cb6dcdfb22add097a49b0b66c7b4308a2ea96c6targetblank8cb6dcda"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/ieoresearch/phenotype_genotype/tree/8cb6dcdfb22add097a49b0b66c7b4308a2ea96c6" target="_blank">8cb6dcd</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/
    Ignored:    analysis/02_integration_phenotype_genotype_sAML1_cache/
    Ignored:    data/
    Ignored:    src/

Untracked files:
    Untracked:  phenotype_genotype.Rproj

Unstaged changes:
    Modified:   analysis/00_malignant_non_malignant_annotations.Rmd
    Modified:   analysis/01_tme_annotation.Rmd
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/02_integration_phenotype_genotype_sAML1.Rmd</code>) and
HTML (<code>docs/02_integration_phenotype_genotype_sAML1.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ieoresearch/phenotype_genotype/blob/8cb6dcdfb22add097a49b0b66c7b4308a2ea96c6/analysis/02_integration_phenotype_genotype_sAML1.Rmd" target="_blank">8cb6dcd</a>
</td>
<td>
Yinxiu Zhan
</td>
<td>
2024-10-23
</td>
<td>
Add sAML1 phenotype genotype analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/ieoresearch/phenotype_genotype/6a3773e38a0f59185e5e84a6dbad7482ef396cca/docs/02_integration_phenotype_genotype_sAML1.html" target="_blank">6a3773e</a>
</td>
<td>
Yinxiu Zhan
</td>
<td>
2024-10-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/ieoresearch/phenotype_genotype/blob/72ca7dceb898f7e2e98d954064a14647a276333a/analysis/02_integration_phenotype_genotype_sAML1.Rmd" target="_blank">72ca7dc</a>
</td>
<td>
Yinxiu Zhan
</td>
<td>
2024-10-21
</td>
<td>
Add sAML1 phenotype genotype analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="objective" class="section level1">
<h1>Objective</h1>
<p>The objective of the analysis is to integrate the genotype data from
the enrichemnt experiment with the expression data. We start with
sAML1.</p>
</div>
<div id="genotype-data" class="section level1">
<h1>Genotype data</h1>
<p>The genotype data has been analyzed using <a
href="https://github.com/dimadatascience/NanoGen">NanoGen</a>, a custom
pipeline designed to call mutations from targeted enrichment data.
NanoGen follows a multi-step process to ensure accurate mutation
detection. It starts by using Blaze to identify cell barcodes, ensuring
that each cell’s data is correctly assigned. It then collapses Unique
Molecular Identifiers (UMIs) to eliminate PCR duplicates and other
biases, allowing for a more accurate representation of the original
sample.</p>
<p>One key feature of NanoGen is its ability to call consensus at the
base level, rather than at the sequence level, providing a more precise
identification of mutations. NanoGen applies a negative binomial model
to distinguish between true mutated reads and background noise,
improving the reliability of mutation calls. To address the issue of
mutated allele dropout, NanoGen imposes a strict threshold: it requires
at least five distinct wild-type (WT) UMIs for a cell to be confidently
classified as wild-type (WT).</p>
<div id="data-exploration" class="section level2">
<h2>Data exploration</h2>
<p>First, we identify and retain only mutations that overlap with the
WES data. For each candidate mutation, we calculate its cellular
fraction. Next, we filter out any genes with fewer than 50 genotyped
cells.</p>
<p>Below, we present the correlation between the cellular fraction,
defined as VAF * 2, in WES and the cellular fraction in the target
enrichment. As demonstrated, the target enrichment effectively
recapitulates the cellular fraction observed in WES.</p>
<div class="plotly html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-782715c6e6448907f7d2" style="width:576px;height:576px;"></div>
<script type="application/json" data-for="htmlwidget-782715c6e6448907f7d2">{"x":{"data":[{"x":[0.44300518134715028,0.30602957906712175,0.6539923954372624,0.74744027303754268,0.98316831683168315,0.54867256637168138,0.88439306358381498],"y":[0.34188034188034189,0.35658914728682173,0.66321243523316065,0.64623955431754876,1,0.56000000000000005,0.73232323232323238],"text":["CSF3R","FLT3","HDAC7","IDH1","RUNX1","SETD2","SRSF2"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,0,0,1)","opacity":1,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(0,0,0,1)"}},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0.2721726421788937,1.0170252537199111],"y":[0.2721726421788937,1.0170252537199111],"text":"","type":"scatter","mode":"lines","line":{"width":1.8897637795275593,"color":"rgba(190,190,190,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":49.872976338729771,"r":7.3059360730593621,"b":37.260273972602747,"l":43.105022831050235},"plot_bgcolor":"rgba(235,235,235,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"title":{"text":"Correlation: 0.95","font":{"color":"rgba(0,0,0,1)","family":"","size":26.567040265670411},"x":0.5,"xref":"paper"},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0.2721726421788937,1.0170252537199111],"tickmode":"array","ticktext":["0.4","0.6","0.8","1.0"],"tickvals":[0.40000000000000002,0.60000000000000009,0.80000000000000004,1],"categoryorder":"array","categoryarray":["0.4","0.6","0.8","1.0"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(255,255,255,1)","gridwidth":0.66417600664176002,"zeroline":false,"anchor":"y","title":{"text":"single cell CF","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0.30897435897435899,1.032905982905983],"tickmode":"array","ticktext":["0.4","0.6","0.8","1.0"],"tickvals":[0.40000000000000002,0.60000000000000009,0.80000000000000004,1],"categoryorder":"array","categoryarray":["0.4","0.6","0.8","1.0"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(255,255,255,1)","gridwidth":0.66417600664176002,"zeroline":false,"anchor":"x","title":{"text":"WES_CF","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.8897637795275593,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"7bda4f03e504":{"x":{},"y":{},"text":{},"type":"scatter"},"7bda4c246226":{"intercept":{},"slope":{}}},"cur_data":"7bda4f03e504","visdat":{"7bda4f03e504":["function (y) ","x"],"7bda4c246226":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-7f2e8bbd475421ad5ec3" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-7f2e8bbd475421ad5ec3">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["CSF3R","FLT3","HDAC7","IDH1","RUNX1","SETD2","SRSF2"],[171,269,172,438,993,62,459],[215,610,91,148,17,51,60],[0.4430051813471503,0.3060295790671217,0.6539923954372624,0.7474402730375427,0.9831683168316832,0.5486725663716814,0.884393063583815],[0.1709401709401709,0.1782945736434109,0.3316062176165803,0.3231197771587744,0.6905829596412556,0.28,0.3661616161616162],[0.3418803418803419,0.3565891472868217,0.6632124352331606,0.6462395543175488,1,0.5600000000000001,0.7323232323232324]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>gene<\/th>\n      <th>MUT<\/th>\n      <th>WT<\/th>\n      <th>ccf<\/th>\n      <th>tumor_f<\/th>\n      <th>wes_ccf<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5]}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="mutation-co-occurence" class="section level2">
<h2>Mutation co-occurence</h2>
<p>Next, we quantified the number of cells with multiple genotyped
mutations, as shown in the tables below. We present the results for
different mutation combinations. For this specific samples, we will look
at these genes since they are the most genotyped: RUNX1, SRSF2, IDH1,
FLT3, CSF3R</p>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-bdb697fcf84443e89835" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-bdb697fcf84443e89835">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["MUT_MUT_MUT","MUT_WT_MUT","WT_MUT_MUT","WT_WT_MUT","WT_WT_WT"],[26,6,68,17,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>FLT3_IDH1_RUNX1<\/th>\n      <th>ncells<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":1}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-e6af24d9f946a256fb7a" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-e6af24d9f946a256fb7a">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["MUT_MUT_MUT","MUT_WT_MUT","WT_MUT_MUT","WT_WT_MUT"],[50,3,77,5]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>FLT3_SRSF2_RUNX1<\/th>\n      <th>ncells<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":1}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-025156185a60de26b921" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-025156185a60de26b921">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["MUT_MUT_MUT","WT_MUT_MUT"],[17,42]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>CSF3R_SRSF2_RUNX1<\/th>\n      <th>ncells<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":1}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-e83d4e6053a8ee756eee" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-e83d4e6053a8ee756eee">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["MUT_MUT_MUT","MUT_WT_MUT","WT_MUT_MUT","WT_WT_MUT"],[20,6,19,7]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>CSF3R_IDH1_RUNX1<\/th>\n      <th>ncells<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":1}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-9495ccb0d834bf177411" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-9495ccb0d834bf177411">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["MUT_MUT_MUT_MUT","MUT_MUT_WT_MUT","WT_MUT_MUT_MUT","WT_MUT_WT_MUT","WT_WT_MUT_MUT"],[9,3,20,7,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>FLT3_SRSF2_IDH1_RUNX1<\/th>\n      <th>ncells<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":1}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="genotype-phenotype-analysis-full-genotype"
class="section level1">
<h1>Genotype phenotype analysis: full genotype</h1>
<p>We begin by analyzing the phenotype-genotype relationships using only
cells with genotyped data for multiple genes. Specifically, we will
focus on the following cases:</p>
<p>1- RUNX1 - IDH1 - FLT3</p>
<p>2- RUNX1 - SRSF2 - FLT3</p>
<p>The reason is that, for these combination, we have the highest number
of genotyped cells for all genes.</p>
<p>For each case, we will examine the impact of adding mutations to the
analysis.</p>
<div id="runx1---idh1---flt3" class="section level2">
<h2>RUNX1 - IDH1 - FLT3</h2>
<p>Describing the mutations in the order RUNX1, IDH1, and FLT3, we
observe the following cell counts: 26 cells with mutations in all three
genes (MMM), 68 cells with mutations in RUNX1 and IDH1 but not FLT3
(MMW), and 17 cells with mutations only in RUNX1 (MWW).</p>
<div id="effect-of-flt3-mutation-mmm-vs-wmm" class="section level3">
<h3>Effect of FLT3 mutation (MMM vs WMM)</h3>
<p>We selected cells where FLT3 is either wild-type or mutated, while
both IDH1 and RUNX1 are mutated. This resulted in the following number
of cells, representing the intersection of genotyped cells and those
passing scRNA-seq filters. The table below presents the number of cells
in each compartment after filtering for cells with available RNA
expression data. Since the majority of cells belong to the tumor
compartment and only a small number are from the TME, we focused our
analysis solely on the tumor compartment.</p>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-9e5c7cbfb9976ef3d6f7" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-9e5c7cbfb9976ef3d6f7">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["tme","tumor","undefined","tme","tumor","undefined"],["MUT_MUT_MUT","MUT_MUT_MUT","MUT_MUT_MUT","WT_MUT_MUT","WT_MUT_MUT","WT_MUT_MUT"],[0,23,0,1,49,2]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>compartment<\/th>\n      <th>mutation_status<\/th>\n      <th>Freq<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":2}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
<p>No genes were identified as significantly differentially expressed
between the two conditions in the differential expression analysis,
likely due to the low number of cells. As a result, we opted to forgo
further comparisons, given the limited cell count.</p>
</div>
</div>
<div id="runx1---srsf2---flt3" class="section level2">
<h2>RUNX1 - SRSF2 - FLT3</h2>
<p>Describing the mutations in the order RUNX1, SRSF2, and FLT3, we
observe the following cell counts: 50 cells with mutations in all three
genes (MMM), 77 cells with mutations in RUNX1 and IDH1 but not FLT3
(MMW), and 5 cells with mutations only in RUNX1 (MWW).</p>
<div id="effect-of-flt3-mutation-mmm-vs-wmm-1" class="section level3">
<h3>Effect of FLT3 mutation (MMM vs WMM)</h3>
<p>We selected cells where FLT3 is either wild-type or mutated, while
both SRSF2 and RUNX1 are mutated. This resulted in the following number
of cells, representing the intersection of genotyped cells and those
passing scRNA-seq filters. The table below presents the number of cells
in each compartment after filtering for cells with available RNA
expression data. Since the majority of cells belong to the tumor
compartment and only a small number are from the TME, we focused our
analysis solely on the tumor compartment.</p>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-2395cee2e1a634da54b8" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-2395cee2e1a634da54b8">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["tme","tumor","undefined","tme","tumor","undefined"],["MUT_MUT_MUT","MUT_MUT_MUT","MUT_MUT_MUT","WT_MUT_MUT","WT_MUT_MUT","WT_MUT_MUT"],[0,32,5,1,51,4]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>compartment<\/th>\n      <th>mutation_status<\/th>\n      <th>Freq<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":2}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
<p>A single gene ‘MGST1’ has been identified as significantly
differentially expressed between the two conditions in the differential
expression analysis, likely due to the low number of cells. As a result,
we opted to forgo further comparisons, given the limited cell count.</p>
</div>
</div>
</div>
<div id="genotype-phenotype-analysis-partial-genotype"
class="section level1">
<h1>Genotype phenotype analysis: partial genotype</h1>
<p>Given the low number of cells, requiring the genotype for all genes
was not feasible. Therefore, we relaxed the genotype criteria and
assumed that the cells acquire mutations in the following order:</p>
<ul>
<li><p>RUNX1 -&gt; SRSF2 -&gt; FLT3</p></li>
<li><p>RUNX1 -&gt; IDH1 -&gt; FLT3</p></li>
</ul>
<p>Therefore, to study the effect of acquiring FLT3 mutations, we can
compare FLT3 MUT cells with the combined group of FLT3 WT + IDH1/SRSF2
MUT</p>
<div id="flt3-mut-vs-flt3-wt-srsf2-mut" class="section level2">
<h2>FLT3 MUT vs FLT3 WT + SRSF2 MUT</h2>
<p>We identified 269 cells with FLT3 mutations and 126 cells with FLT3
WT and SRSF2 mutations. The table below presents the number of cells in
each compartment after filtering for cells with available RNA expression
data. Since the majority of cells belong to the tumor compartment and
only a small number are from the TME, we focused our analysis solely on
the tumor compartment.</p>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-d1e2ced4dfaae21ba9e9" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-d1e2ced4dfaae21ba9e9">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["tme","tumor","undefined","tme","tumor","undefined"],["MUT","MUT","MUT","WT_MUT","WT_MUT","WT_MUT"],[0,207,17,1,89,8]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>compartment<\/th>\n      <th>mutation_status<\/th>\n      <th>Freq<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":2}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
<p>We conducted differential gene expression analysis between the two
groups and followed it with reactome pathway enrichment analysis. The
dot plot below, along with the accompanying table, presents the enriched
terms.</p>
<p><strong>Enriched in FLT3 Mutated</strong></p>
<p><img src="figure/02_integration_phenotype_genotype_sAML1.Rmd/unnamed-chunk-19-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ieoresearch/phenotype_genotype/blob/6a3773e38a0f59185e5e84a6dbad7482ef396cca/docs/figure/02_integration_phenotype_genotype_sAML1.Rmd/unnamed-chunk-19-1.png" target="_blank">6a3773e</a>
</td>
<td>
Yinxiu Zhan
</td>
<td>
2024-10-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-594d15185baacc5d98b1" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-594d15185baacc5d98b1">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[[0.02074219206878181,0.02074219206878181,0.02074219206878181,0.02074219206878181,0.02074219206878181,0.02497914353881877],["Negative regulation of FLT3","Growth hormone receptor signaling","Pyruvate metabolism","Interleukin-7 signaling","FLT3 Signaling","Pyruvate metabolism and Citric Acid (TCA) cycle"],["SOCS2","SOCS2","LDHB","SOCS2","SOCS2","LDHB"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>p.adjust<\/th>\n      <th>Description<\/th>\n      <th>geneID<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
<p><strong>Enriched in FLT3 WT + SRSF2 Mutated</strong></p>
<p><img src="figure/02_integration_phenotype_genotype_sAML1.Rmd/unnamed-chunk-20-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ieoresearch/phenotype_genotype/blob/6a3773e38a0f59185e5e84a6dbad7482ef396cca/docs/figure/02_integration_phenotype_genotype_sAML1.Rmd/unnamed-chunk-20-1.png" target="_blank">6a3773e</a>
</td>
<td>
Yinxiu Zhan
</td>
<td>
2024-10-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-979208747365879458f5" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-979208747365879458f5">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[[0.001637166054056164,0.02508591244773577],["Interleukin-4 and Interleukin-13 signaling","Signaling by Interleukins"],["ANXA1/MCL1/HGF/CEBPD","ANXA1/MCL1/HGF/IL1RAP/CEBPD"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>p.adjust<\/th>\n      <th>Description<\/th>\n      <th>geneID<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="flt3-mut-vs-flt3-wt-idh1-mut" class="section level2">
<h2>FLT3 MUT vs FLT3 WT + IDH1 MUT</h2>
<p>We identified 269 cells with FLT3 mutations and 143 cells with FLT3
WT and IDH1 mutations. The table below presents the number of cells in
each compartment after filtering for cells with available RNA expression
data. Since the majority of cells belong to the tumor compartment and
only a small number are from the TME, we focused our analysis solely on
the tumor compartment.</p>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-d7bf592c970bd4b32817" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-d7bf592c970bd4b32817">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["tme","tumor","undefined","tme","tumor","undefined"],["MUT","MUT","MUT","WT_MUT","WT_MUT","WT_MUT"],[0,207,17,1,116,2]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>compartment<\/th>\n      <th>mutation_status<\/th>\n      <th>Freq<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":2}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
<p>We conducted differential gene expression analysis between the two
groups and followed it with reactome pathway enrichment analysis. The
dot plot below, along with the accompanying table, presents the enriched
terms.</p>
<p><strong>Enriched in FLT3 Mutated</strong></p>
<p><img src="figure/02_integration_phenotype_genotype_sAML1.Rmd/unnamed-chunk-24-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ieoresearch/phenotype_genotype/blob/6a3773e38a0f59185e5e84a6dbad7482ef396cca/docs/figure/02_integration_phenotype_genotype_sAML1.Rmd/unnamed-chunk-24-1.png" target="_blank">6a3773e</a>
</td>
<td>
Yinxiu Zhan
</td>
<td>
2024-10-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-28034cbf781656ae8e53" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-28034cbf781656ae8e53">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[[1.823275483630249e-07,1.823275483630249e-07,1.823275483630249e-07,1.823275483630249e-07,1.823275483630249e-07,1.823275483630249e-07,2.096302639425337e-07,2.096302639425337e-07,2.866364190719586e-07,2.866364190719586e-07,2.866364190719586e-07,2.866364190719586e-07,2.950566250236302e-07,2.950566250236302e-07,3.078175600942697e-07,3.078175600942697e-07,3.078175600942697e-07,6.933047683142984e-07,9.114772357991963e-07,1.598013621157324e-06,1.598013621157324e-06,4.674338671912099e-06,6.387431907870663e-06,8.58004556119848e-06,2.540991585033886e-05,2.692410624407165e-05,3.893110109256476e-05,4.310322432322591e-05,5.296795483944963e-05,7.213563994747552e-05,0.0002623125638127193,0.002571271568752739,0.002641308186607973,0.003618030188783392,0.003618030188783392,0.003696087656412867,0.0294241854200935,0.03233279856872118],["Peptide chain elongation","Viral mRNA Translation","Eukaryotic Translation Elongation","Selenocysteine synthesis","Eukaryotic Translation Termination","Nonsense Mediated Decay (NMD) independent of the Exon Junction Complex (EJC)","Formation of a pool of free 40S subunits","Response of EIF2AK4 (GCN2) to amino acid deficiency","Regulation of expression of SLITs and ROBOs","L13a-mediated translational silencing of Ceruloplasmin expression","SRP-dependent cotranslational protein targeting to membrane","GTP hydrolysis and joining of the 60S ribosomal subunit","Nonsense-Mediated Decay (NMD)","Nonsense Mediated Decay (NMD) enhanced by the Exon Junction Complex (EJC)","Selenoamino acid metabolism","Eukaryotic Translation Initiation","Cap-dependent Translation Initiation","Influenza Viral RNA Transcription and Replication","Signaling by ROBO receptors","Influenza Infection","Cellular response to starvation","Major pathway of rRNA processing in the nucleolus and cytosol","rRNA processing in the nucleus and cytosol","rRNA processing","SARS-CoV-1 modulates host translation machinery","Metallothioneins bind metals","Metabolism of amino acids and derivatives","SARS-CoV-1-host interactions","Response to metal ions","Translation","SARS-CoV-1 Infection","SARS-CoV-2 modulates host translation machinery","Formation of the ternary complex, and subsequently, the 43S complex","Translation initiation complex formation","Ribosomal scanning and start codon recognition","Activation of the mRNA upon binding of the cap-binding complex and eIFs, and subsequent binding to 43S","SARS-CoV Infections","Host Interactions of HIV factors"],["RPL4/RPSA/RPLP1/RPS16/RPL19/RPS11/RPL37A","RPL4/RPSA/RPLP1/RPS16/RPL19/RPS11/RPL37A","RPL4/RPSA/RPLP1/RPS16/RPL19/RPS11/RPL37A","RPL4/RPSA/RPLP1/RPS16/RPL19/RPS11/RPL37A","RPL4/RPSA/RPLP1/RPS16/RPL19/RPS11/RPL37A","RPL4/RPSA/RPLP1/RPS16/RPL19/RPS11/RPL37A","RPL4/RPSA/RPLP1/RPS16/RPL19/RPS11/RPL37A","RPL4/RPSA/RPLP1/RPS16/RPL19/RPS11/RPL37A","PSMB9/RPL4/RPSA/RPLP1/RPS16/RPL19/RPS11/RPL37A","RPL4/RPSA/RPLP1/RPS16/RPL19/RPS11/RPL37A","RPL4/RPSA/RPLP1/RPS16/RPL19/RPS11/RPL37A","RPL4/RPSA/RPLP1/RPS16/RPL19/RPS11/RPL37A","RPL4/RPSA/RPLP1/RPS16/RPL19/RPS11/RPL37A","RPL4/RPSA/RPLP1/RPS16/RPL19/RPS11/RPL37A","RPL4/RPSA/RPLP1/RPS16/RPL19/RPS11/RPL37A","RPL4/RPSA/RPLP1/RPS16/RPL19/RPS11/RPL37A","RPL4/RPSA/RPLP1/RPS16/RPL19/RPS11/RPL37A","RPL4/RPSA/RPLP1/RPS16/RPL19/RPS11/RPL37A","PSMB9/RPL4/RPSA/RPLP1/RPS16/RPL19/RPS11/RPL37A","RPL4/RPSA/RPLP1/RPS16/RPL19/RPS11/RPL37A","RPL4/RPSA/RPLP1/RPS16/RPL19/RPS11/RPL37A","RPL4/RPSA/RPLP1/RPS16/RPL19/RPS11/RPL37A","RPL4/RPSA/RPLP1/RPS16/RPL19/RPS11/RPL37A","RPL4/RPSA/RPLP1/RPS16/RPL19/RPS11/RPL37A","HNRNPA1/RPSA/RPS16/RPS11","MT2A/MT1X/MT1E","PSMB9/RPL4/RPSA/RPLP1/RPS16/RPL19/RPS11/RPL37A","HNRNPA1/NPM1/RPSA/RPS16/RPS11","MT2A/MT1X/MT1E","RPL4/RPSA/RPLP1/RPS16/RPL19/RPS11/RPL37A","HNRNPA1/NPM1/RPSA/RPS16/RPS11","RPSA/RPS16/RPS11","RPSA/RPS16/RPS11","RPSA/RPS16/RPS11","RPSA/RPS16/RPS11","RPSA/RPS16/RPS11","HNRNPA1/NPM1/RPSA/RPS16/RPS11","HMGA1/PSMB9/NPM1"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>p.adjust<\/th>\n      <th>Description<\/th>\n      <th>geneID<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
<p><strong>Enriched in FLT3 WT + IDH1 Mutated</strong></p>
<p><img src="figure/02_integration_phenotype_genotype_sAML1.Rmd/unnamed-chunk-25-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-1">
Past versions of unnamed-chunk-25-1.png
</button>
</p>
<div id="fig-unnamed-chunk-25-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ieoresearch/phenotype_genotype/blob/6a3773e38a0f59185e5e84a6dbad7482ef396cca/docs/figure/02_integration_phenotype_genotype_sAML1.Rmd/unnamed-chunk-25-1.png" target="_blank">6a3773e</a>
</td>
<td>
Yinxiu Zhan
</td>
<td>
2024-10-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-98c18d0c24887fb486df" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-98c18d0c24887fb486df">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[[0.005366734305690184,0.01422020120783447,0.01958842980266219],["Interleukin-4 and Interleukin-13 signaling","Neutrophil degranulation","NGF-stimulated transcription"],["ANXA1/JUNB/FOS/HGF/MCL1","MGST1/GLIPR1/CD44/IDH1/PLD1/BIN2/SERPINB1/FTL","JUNB/FOS/JUND"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>p.adjust<\/th>\n      <th>Description<\/th>\n      <th>geneID<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="flt3-mut-vs-flt3-wt" class="section level2">
<h2>FLT3 MUT vs FLT3 WT</h2>
<p>We identified 269 cells with FLT3 mutations and 610 cells with FLT3
WT. The table below presents the number of cells in each compartment
after filtering for cells with available RNA expression data. Since the
majority of cells belong to the tumor compartment and only a small
number are from the TME, we focused our analysis solely on the tumor
compartment.</p>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-4767b6395d1e4bfe93a0" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-4767b6395d1e4bfe93a0">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["tme","tumor","undefined","tme","tumor","undefined"],["MUT","MUT","MUT","WT","WT","WT"],[0,207,17,3,492,16]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>compartment<\/th>\n      <th>mutation_status<\/th>\n      <th>Freq<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":2}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
<p>We conducted differential gene expression analysis between the two
groups and followed it with reactome pathway enrichment analysis. The
dot plot below, along with the accompanying table, presents the enriched
terms.</p>
<p><strong>Enriched in FLT3 Mutated</strong></p>
<p><img src="figure/02_integration_phenotype_genotype_sAML1.Rmd/unnamed-chunk-29-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-29-1">
Past versions of unnamed-chunk-29-1.png
</button>
</p>
<div id="fig-unnamed-chunk-29-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ieoresearch/phenotype_genotype/blob/6a3773e38a0f59185e5e84a6dbad7482ef396cca/docs/figure/02_integration_phenotype_genotype_sAML1.Rmd/unnamed-chunk-29-1.png" target="_blank">6a3773e</a>
</td>
<td>
Yinxiu Zhan
</td>
<td>
2024-10-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-54b94b9422a61b8eafa4" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-54b94b9422a61b8eafa4">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[[8.417480041761537e-06,1.268276918803573e-05,0.003995566925940004,0.003995566925940004,0.005902505034748831,0.01479905541544171,0.0243253577309799,0.02536340515453039,0.02536340515453039,0.02536340515453039,0.04116429652585225,0.04318383958284646,0.04349494884442977,0.04349494884442977,0.04349494884442977,0.04349494884442977,0.04758591037701311,0.04758591037701311,0.04758591037701311,0.04758591037701311],["Metallothioneins bind metals","Response to metal ions","Host Interactions of HIV factors","G1/S Transition","Mitotic G1 phase and G1/S transition","TNFR2 non-canonical NF-kB pathway","Synthesis of DNA","HIV Infection","Mitotic Anaphase","Mitotic Metaphase and Anaphase","Orc1 removal from chromatin","S Phase","Nuclear events mediated by NFE2L2","Cell Cycle Checkpoints","Deregulated CDK5 triggers multiple neurodegenerative pathways in Alzheimer's disease models","Neurodegenerative Diseases","DNA Replication","Defective Intrinsic Pathway for Apoptosis","Separation of Sister Chromatids","Switching of origins to a post-replicative state"],["MT2A/MT1X/MT1E/MT1F","MT2A/MT1X/MT1E/MT1F","HMGA1/PSMB9/RAN/PSMB8/SLC25A5","PCNA/TYMS/PSMB9/MCM7/PSMB8","PCNA/TYMS/PSMB9/MCM7/PSMB8","LTB/TNFRSF14/PSMB9/PSMB8","PCNA/PSMB9/MCM7/PSMB8","HMGA1/PSMB9/RAN/PSMB8/SLC25A5","PSMB9/RAN/CENPU/PSMB8/DYNLL1","PSMB9/RAN/CENPU/PSMB8/DYNLL1","PSMB9/MCM7/PSMB8","PCNA/PSMB9/MCM7/PSMB8","PRDX1/PSMB9/PSMB8","PSMB9/MCM7/CENPU/PSMB8/DYNLL1","PRDX1/SOD2","PRDX1/SOD2","PCNA/PSMB9/MCM7/PSMB8","PRDX1/SOD2","PSMB9/CENPU/PSMB8/DYNLL1","PSMB9/MCM7/PSMB8"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>p.adjust<\/th>\n      <th>Description<\/th>\n      <th>geneID<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
<p><strong>Enriched in FLT3 WT</strong></p>
<p><img src="figure/02_integration_phenotype_genotype_sAML1.Rmd/unnamed-chunk-30-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-30-1">
Past versions of unnamed-chunk-30-1.png
</button>
</p>
<div id="fig-unnamed-chunk-30-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/ieoresearch/phenotype_genotype/blob/6a3773e38a0f59185e5e84a6dbad7482ef396cca/docs/figure/02_integration_phenotype_genotype_sAML1.Rmd/unnamed-chunk-30-1.png" target="_blank">6a3773e</a>
</td>
<td>
Yinxiu Zhan
</td>
<td>
2024-10-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-803833d68677a5c6e327" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-803833d68677a5c6e327">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[[0.001818309228627809,0.01450007151688468,0.03655298675146396],["Interleukin-4 and Interleukin-13 signaling","NGF-stimulated transcription","Nuclear Events (kinase and transcription factor activation)"],["FOS/ANXA1/JUNB/MCL1/HGF","FOS/JUNB/JUND","FOS/JUNB/JUND"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>p.adjust<\/th>\n      <th>Description<\/th>\n      <th>geneID<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="idh1-mut-vs-idh1-wt" class="section level2">
<h2>IDH1 MUT vs IDH1 WT</h2>
<p>We identified 438 cells with IDH1 mutations and 148 cells with IDH1
WT. The table below presents the number of cells in each compartment
after filtering for cells with available RNA expression data. Since the
majority of cells belong to the tumor compartment and only a small
number are from the TME, we focused our analysis solely on the tumor
compartment.</p>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-a83b520f475d9408a049" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-a83b520f475d9408a049">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["tme","tumor","undefined","tme","tumor","undefined"],["MUT","MUT","MUT","WT","WT","WT"],[5,356,12,4,115,5]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>compartment<\/th>\n      <th>mutation_status<\/th>\n      <th>Freq<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":2}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
<p>No differenetially expressed genes have been detected</p>
</div>
<div id="idh1-mut-vs-idh1-wt-in-flt3-wt" class="section level2">
<h2>IDH1 MUT vs IDH1 WT in FLT3 WT</h2>
<p>We identified 143 cells with IDH1 mutations and 33 cells with IDH1 WT
in FLT3 WT condition. The table below presents the number of cells in
each compartment after filtering for cells with available RNA expression
data. Since the majority of cells belong to the tumor compartment and
only a small number are from the TME, we focused our analysis solely on
the tumor compartment.</p>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-0db7182e882e2ed72d4a" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-0db7182e882e2ed72d4a">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["tme","tumor","undefined","tme","tumor","undefined"],["MUT_WT","MUT_WT","MUT_WT","WT_WT","WT_WT","WT_WT"],[1,116,2,0,27,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>compartment<\/th>\n      <th>mutation_status<\/th>\n      <th>Freq<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":2}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
<p>No differentially expressed genes have been detected</p>
</div>
<div id="srsf2-mut-vs-srsf2-wt" class="section level2">
<h2>SRSF2 MUT vs SRSF2 WT</h2>
<p>We identified 459 cells with SRSF2 mutations and 60 cells with IDH1
WT. The table below presents the number of cells in each compartment
after filtering for cells with available RNA expression data. Since the
majority of cells belong to the tumor compartment and only a small
number are from the TME, we focused our analysis solely on the tumor
compartment.</p>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-65f9dfa7855fc4e94c22" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-65f9dfa7855fc4e94c22">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["tme","tumor","undefined","tme","tumor","undefined"],["MUT","MUT","MUT","WT","WT","WT"],[6,303,54,6,31,6]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>compartment<\/th>\n      <th>mutation_status<\/th>\n      <th>Freq<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":2}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
<p>No differenetially expressed genes have been detected</p>
</div>
<div id="srsf2-mut-vs-srsf2-wt-in-flt3-wt-condition"
class="section level2">
<h2>SRSF2 MUT vs SRSF2 WT in FLT3 WT condition</h2>
<p>We identified 126 cells with SRSF2 mutations and 12 cells with IDH1
WT. The table below presents the number of cells in each compartment
after filtering for cells with available RNA expression data. Since the
majority of cells belong to the tumor compartment and only a small
number are from the TME, we focused our analysis solely on the tumor
compartment.</p>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-90eea304ff52992fa43a" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-90eea304ff52992fa43a">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["tme","tumor","undefined","tme","tumor","undefined"],["MUT_WT","MUT_WT","MUT_WT","WT_WT","WT_WT","WT_WT"],[1,89,8,0,8,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>compartment<\/th>\n      <th>mutation_status<\/th>\n      <th>Freq<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":2}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
<p>No differentially expressed genes have been detected</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.3.1 (2023-06-16)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ReactomePA_1.44.0     clusterProfiler_4.8.3 org.Hs.eg.db_3.17.0  
 [4] AnnotationDbi_1.62.2  IRanges_2.34.1        S4Vectors_0.38.1     
 [7] Biobase_2.60.0        BiocGenerics_0.46.0   plotly_4.10.2        
[10] maftools_2.16.0       mclust_6.0.0          SeuratObject_4.1.3   
[13] Seurat_4.3.0.1        lubridate_1.9.2       forcats_1.0.0        
[16] stringr_1.5.0         dplyr_1.1.3           purrr_1.0.2          
[19] readr_2.1.4           tidyr_1.3.0           tibble_3.2.1         
[22] ggplot2_3.4.2         tidyverse_2.0.0      

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.21        splines_4.3.1           later_1.3.1            
  [4] ggplotify_0.1.2         bitops_1.0-7            polyclip_1.10-4        
  [7] graph_1.78.0            lifecycle_1.0.3         rprojroot_2.0.3        
 [10] globals_0.16.2          lattice_0.21-8          MASS_7.3-60            
 [13] crosstalk_1.2.0         magrittr_2.0.3          limma_3.56.2           
 [16] sass_0.4.7              rmarkdown_2.24          jquerylib_0.1.4        
 [19] yaml_2.3.7              httpuv_1.6.11           sctransform_0.3.5      
 [22] sp_2.0-0                spatstat.sparse_3.0-2   reticulate_1.32.0      
 [25] cowplot_1.1.1           pbapply_1.7-2           DBI_1.1.3              
 [28] RColorBrewer_1.1-3      abind_1.4-5             zlibbioc_1.46.0        
 [31] Rtsne_0.16              ggraph_2.1.0            RCurl_1.98-1.12        
 [34] yulab.utils_0.0.9       rappdirs_0.3.3          tweenr_2.0.2           
 [37] git2r_0.32.0            GenomeInfoDbData_1.2.10 enrichplot_1.20.1      
 [40] ggrepel_0.9.3           irlba_2.3.5.1           listenv_0.9.0          
 [43] spatstat.utils_3.0-3    tidytree_0.4.5          reactome.db_1.84.0     
 [46] goftest_1.2-3           spatstat.random_3.1-6   fitdistrplus_1.1-11    
 [49] parallelly_1.36.0       leiden_0.4.3            codetools_0.2-19       
 [52] DT_0.29                 ggforce_0.4.1           DOSE_3.26.1            
 [55] DNAcopy_1.74.1          tidyselect_1.2.0        aplot_0.2.0            
 [58] farver_2.1.1            viridis_0.6.4           matrixStats_1.0.0      
 [61] spatstat.explore_3.2-3  jsonlite_1.8.7          tidygraph_1.2.3        
 [64] ellipsis_0.3.2          progressr_0.14.0        ggridges_0.5.4         
 [67] survival_3.5-5          tools_4.3.1             treeio_1.24.3          
 [70] ica_1.0-3               Rcpp_1.0.11             glue_1.6.2             
 [73] gridExtra_2.3           xfun_0.40               qvalue_2.32.0          
 [76] GenomeInfoDb_1.36.3     withr_2.5.0             fastmap_1.1.1          
 [79] fansi_1.0.4             digest_0.6.33           gridGraphics_0.5-1     
 [82] timechange_0.2.0        R6_2.5.1                mime_0.12              
 [85] colorspace_2.1-0        scattermore_1.2         GO.db_3.17.0           
 [88] tensor_1.5              spatstat.data_3.0-1     RSQLite_2.3.1          
 [91] utf8_1.2.3              generics_0.1.3          data.table_1.14.8      
 [94] graphlayouts_1.0.0      httr_1.4.7              htmlwidgets_1.6.2      
 [97] scatterpie_0.2.1        graphite_1.46.0         whisker_0.4.1          
[100] uwot_0.1.16             pkgconfig_2.0.3         gtable_0.3.4           
[103] blob_1.2.4              workflowr_1.7.1         lmtest_0.9-40          
[106] XVector_0.40.0          shadowtext_0.1.2        htmltools_0.5.6        
[109] fgsea_1.26.0            scales_1.2.1            png_0.1-8              
[112] ggfun_0.1.2             knitr_1.44              rstudioapi_0.15.0      
[115] tzdb_0.4.0              reshape2_1.4.4          nlme_3.1-162           
[118] cachem_1.0.8            zoo_1.8-12              KernSmooth_2.23-22     
[121] parallel_4.3.1          miniUI_0.1.1.1          HDO.db_0.99.1          
[124] pillar_1.9.0            grid_4.3.1              vctrs_0.6.3            
[127] RANN_2.6.1              promises_1.2.1          xtable_1.8-4           
[130] cluster_2.1.4           evaluate_0.21           cli_3.6.1              
[133] compiler_4.3.1          rlang_1.1.1             crayon_1.5.2           
[136] future.apply_1.11.0     labeling_0.4.3          plyr_1.8.8             
[139] fs_1.6.3                stringi_1.7.12          viridisLite_0.4.2      
[142] deldir_1.0-9            BiocParallel_1.34.2     munsell_0.5.0          
[145] Biostrings_2.68.1       lazyeval_0.2.2          spatstat.geom_3.2-5    
[148] GOSemSim_2.26.1         Matrix_1.5-3            hms_1.1.3              
[151] patchwork_1.1.3         bit64_4.0.5             future_1.33.0          
[154] KEGGREST_1.40.0         shiny_1.7.5             highr_0.10             
[157] ROCR_1.0-11             igraph_1.4.3            memoise_2.0.1          
[160] bslib_0.5.1             ggtree_3.8.2            fastmatch_1.1-4        
[163] bit_4.0.5               downloader_0.4          gson_0.1.0             
[166] ape_5.7-1              </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
