---
title: "Genotype phenotype AML5"
author: 
- Marcello Del Corvo
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  workflowr::wflow_html:
    toc_depth: 4
editor_options:
  chunk_output_type: console
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo       = FALSE,
                      message    = FALSE,
                      warning    = FALSE,
                      cache      = FALSE,
                      autodep    = TRUE,
                      fig.align  = 'center',
                      fig.width  = 6,
                      fig.height = 6)
```

```{r}
suppressPackageStartupMessages({
  library(tidyverse)
  library(Seurat)
  library(SeuratObject)
  library(ggplot2)
  library(dplyr)
  library(tidyr)
  library(mclust)
  library(maftools)
  library(ggplot2)
  library(plotly)
  library(AnnotationDbi)
  library(org.Hs.eg.db)
  library(clusterProfiler)
  library(ReactomePA)
  source('src/functions.R')
})

```

# Objective

Genotype data integration of ONT enrichment experiment with 10X expression data for AML5 sample


# Genotype data

The genotype data has been analyzed using [NanoGen](https://github.com/dimadatascience/NanoGen), a custom pipeline designed to call mutations from targeted enrichment data. NanoGen follows a multi-step process to ensure accurate mutation detection. It starts by using Blaze to identify cell barcodes, ensuring that each cellâ€™s data is correctly assigned. It then collapses Unique Molecular Identifiers (UMIs) to eliminate PCR duplicates and other biases, allowing for a more accurate representation of the original sample.

One key feature of NanoGen is its ability to call consensus at the base level, rather than at the sequence level, providing a more precise identification of mutations. NanoGen applies a negative binomial model to distinguish between true mutated reads and background noise, improving the reliability of mutation calls. To address the issue of mutated allele dropout, NanoGen imposes a strict threshold: it requires at least five distinct wild-type (WT) UMIs for a cell to be confidently classified as wild-type (WT).

## Data exploration

First, we identify and retain only mutations that overlap with the WES data. For each candidate mutation, we calculate its cellular fraction. Next, we filter out any genes with fewer than 50 genotyped cells.

```{r}
genotype_file = "data/enrichment/AML5.csv"
wes_file = "data/S34030_AML5_D.small_mutations.cancervar.escat.maf"
sample_id = "AML5"
```

```{r, results=FALSE}
# Read and filter genotyped cells
genotype = read.csv(genotype_file, row.names = 1)
genotype = genotype[genotype$genotype!="", ]
genotype['total_umis'] = (genotype[,'WT'] + genotype[,'MUT'] + genotype[,'MIS'])
genotype$alt <- ifelse(grepl('-', genotype$alt), '-', genotype$alt)
genotype['id'] = paste0(genotype[,'gene'], genotype[,'start'], genotype[,'alt'])

variants = data.frame(read.maf(wes_file)@data)

variants['id'] = paste0(variants[,'Hugo_Symbol'] , variants[,'Start_Position'] , variants[,'Tumor_Seq_Allele2'])
genotype = as_tibble(genotype[genotype$id %in% (variants$id), ])

# Calculate ccf and filter lowly genotyped mutations
ccf = genotype %>%
  group_by(gene, genotype) %>%            # Group by 'gene' and 'genotype'
  summarise(count = n(), .groups = 'drop') %>%  # Count occurrences of each 'genotype'
  pivot_wider(names_from = genotype,
              values_from = count,
              values_fill = 0)  # Pivot to wide format, filling missing values with 0

ccf = ccf[(ccf$MUT + ccf$WT) > 50,]
ccf['ccf'] = ccf[,'MUT'] /(ccf[,'MUT'] + ccf[,'WT']) 

# Add wes
subset <- variants %>%
  filter(id %in% genotype$id) %>%
  select(Hugo_Symbol, tumor_f)

ccf <- ccf %>%
  left_join(subset, by = c("gene" = "Hugo_Symbol"))  %>%
  mutate(wes_ccf = pmin(tumor_f * 2, 1))  # Ensure values > 1 are set to 1

# Calculate correlation
correlation <- cor(ccf$ccf, ccf$wes_ccf, use = "complete.obs")
correlation_title <- sprintf("Correlation: %.2f", correlation)



# Create the scatter plot
p <- ggplot(ccf, aes(x = ccf, y = wes_ccf, text = gene)) +
  geom_point() +
  geom_abline(slope = 1, intercept = 0, linetype = "dashed", color = "grey") +
  ggtitle(correlation_title) +
  labs(x = "single cell CF", y = "WES_CF") +
  theme(plot.title = element_text(hjust = 0.5, size = 20))

# Convert to interactive plotly plot
fig <- ggplotly(p, tooltip = c("text"))
```

Below, we present the correlation between the cellular fraction, defined as VAF * 2, in WES  and the cellular fraction in the target enrichment. As demonstrated, the target enrichment effectively recapitulates the cellular fraction observed in WES.

```{r}
fig
```


```{r}
DT::datatable(
  ccf,
  extensions = 'Buttons',
  options = list(
    dom = 'Blfrtip',
    buttons = c('copy', 'csv', 'excel', 'pdf'),
    lengthMenu = list(c(10, 25, 50,-1),
                      c(10, 25, 50, "All"))
  ),
  rownames = FALSE
)
```


## Mutation co-occurence  

Next, we quantified the number of cells with multiple genotyped mutations, as shown in the tables below. We present the results for different mutation combinations. For this specific samples, we will look at these genes since they are the most genotyped: RUNX1, SRSF2, EZH2

```{r}
# Define the genes of interest
gene_interest <- c('EZH2', 'SRSF2', 'RUNX1')
co_occurence <- get_coccurence(genotype_file, variants, gene_interest)
DT::datatable(
  co_occurence,
  extensions = 'Buttons',
  options = list(
    dom = 'Blfrtip',
    buttons = c('copy', 'csv', 'excel', 'pdf'),
    lengthMenu = list(c(10, 25, 50,-1),
                      c(10, 25, 50, "All"))
  ),
  rownames = FALSE
)


```

```{r}
# Read RNAseq
srat = readRDS("data/01_tme_annotation.rds")
srat$cellid = unlist(lapply(str_split(names(srat$orig.ident), "-"), "[[", 1))
```


# Genotype phenotype analysis: partial genotype

Given the low number of cells, requiring the genotype for all genes was not feasible. We studied therefore only single gene genotypes, thus the effect of mutation on a single gene.

## SRSF2 MUT vs  SRSF2 WT 

We identified 571 cells with SRSF2 mutations and 46 cells with SRSF2 WT.  The table below presents the number of cells in each compartment after filtering for cells with available RNA expression data. We focused our analysis solely on the tumor compartment.

```{r test}
# Extract mutated FRYL cells
gene_interest <- c('SRSF2')
mutation_status1 = c('MUT')
subset = get_cellid_full_genotype(genotype_file, variants, gene_interest)
cells1 = subset$cell[subset$SRSF2 == "MUT"]

mutation_status2 = c("WT")
cells2 = subset$cell[subset$SRSF2 == "WT"]

rna = subset(srat, subset = sample_id == sample_id & cellid %in% c(cells1, cells2))
rna$mutation_status = paste(mutation_status1, collapse = "_")
rna$mutation_status[rna$cellid %in% cells2 ] = paste(mutation_status2, collapse = "_")
```


```{r}
DT::datatable(
  data.frame(table(rna@meta.data[, c("compartment", "mutation_status")])),
  extensions = 'Buttons',
  options = list(
    dom = 'Blfrtip',
    buttons = c('copy', 'csv', 'excel', 'pdf'),
    lengthMenu = list(c(10, 25, 50,-1),
                      c(10, 25, 50, "All"))
  ),
  rownames = FALSE
)
```


No differenetially expressed genes have been detected


## RUNX1 MUT vs  RUNX1 WT 

We identified 954 cells with RUNX1 mutations and 17 cells with RUNX1 WT.  The table below presents the number of cells in each compartment after filtering for cells with available RNA expression data. We focused our analysis solely on the tumor compartment.

```{r}
# Extract mutated FRYL cells
gene_interest <- c('RUNX1')
mutation_status1 = c('MUT')
subset = get_cellid_full_genotype(genotype_file, variants, gene_interest)
cells1 = subset$cell[subset$RUNX1 == "MUT"]

mutation_status2 = c("WT")
cells2 = subset$cell[subset$RUNX1 == "WT"]

rna = subset(srat, subset = sample_id == sample_id & cellid %in% c(cells1, cells2))
rna$mutation_status = paste(mutation_status1, collapse = "_")
rna$mutation_status[rna$cellid %in% cells2 ] = paste(mutation_status2, collapse = "_")
```


```{r}
DT::datatable(
  data.frame(table(rna@meta.data[, c("compartment", "mutation_status")])),
  extensions = 'Buttons',
  options = list(
    dom = 'Blfrtip',
    buttons = c('copy', 'csv', 'excel', 'pdf'),
    lengthMenu = list(c(10, 25, 50,-1),
                      c(10, 25, 50, "All"))
  ),
  rownames = FALSE
)
```


No differenetially expressed genes have been detected



## EZH2 MUT vs  EZH2 WT 

We identified 84 cells with EZH2 mutations and 17 cells with EZH2 WT.  The table below presents the number of cells in each compartment after filtering for cells with available RNA expression data. We focused our analysis solely on the tumor compartment.

```{r}
# Extract mutated FRYL cells
gene_interest <- c('EZH2')
mutation_status1 = c('MUT')
subset = get_cellid_full_genotype(genotype_file, variants, gene_interest)
cells1 = subset$cell[subset$EZH2 == "MUT"]

mutation_status2 = c("WT")
cells2 = subset$cell[subset$EZH2 == "WT"]

rna = subset(srat, subset = sample_id == sample_id & cellid %in% c(cells1, cells2))
rna$mutation_status = paste(mutation_status1, collapse = "_")
rna$mutation_status[rna$cellid %in% cells2 ] = paste(mutation_status2, collapse = "_")
```


```{r}
DT::datatable(
  data.frame(table(rna@meta.data[, c("compartment", "mutation_status")])),
  extensions = 'Buttons',
  options = list(
    dom = 'Blfrtip',
    buttons = c('copy', 'csv', 'excel', 'pdf'),
    lengthMenu = list(c(10, 25, 50,-1),
                      c(10, 25, 50, "All"))
  ),
  rownames = FALSE
)
```



No differenetially expressed genes have been detected

